core/src/toga/app.py

```diff
@@ -127,43 +127,45 @@ def __init__(
         title: str | None = None,
```

Comment:
If a default title of `None` is passed, this currently causes a segfault on macOS on startup:
```
2023-08-05 01:01:45.807 Python[65627:1476985] *** Assertion failure in -[TogaWindow setTitle:], NSWindow.m:2771
2023-08-05 01:01:45.809 Python[65627:1476985] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Invalid parameter not satisfying: aString != nil'
```

Reply:
Good catch. I can't think of any good reason that we should allow `None`; a window always needs a title.

Reply:
Actually, from an implementation perspective, it's helpful to have `None` as a "use default value" marker - but by the time we're actually setting the title, it definitely shouldn't be None. The bug here was that constructing the window doesn't use the `title` accessor.

Reply:
Hm, using `None` as default marker mostly makes sense IMO if we expect different defaults depending on the implementation layer. Is this the case? Otherwise, we can just specify the default in the argument, for example an empty string.

Btw, `toga.Window` currently uses `title = "Toga"` as a default.

Reply:
The "Toga" title is intentional; MainWindow and Window have different default titles (MainWindow defaults to the app formal name; Window defaults to "Toga")

There's 2 additional complications here:

1. The default for MainWindow is dynamic - i.e we can't specify a constant, because the value changes at runtime
2. We also need to support the default value when handling `my_window.title = None`. 

With those constraints, using `None` as the default seemed the cleanest approach - there's a single point of entry for evaluating the default (in the call to `set_title()`), and a single property to override to get the default value.

Reply:
Thanks for the explanation! In that case, I completely agree with the choice of None.